### ISSUE TITLE
Is there a detailed list of variables?

### ISSUE THREAD

**Issue by JAK0723** [PetterKraabol/Twitch-Chat-Downloader#66]
I was looking into a way to change the output format so that that the filename would include details such as the video title. However, the [custom formats page](https://github.com/PetterKraabol/Twitch-Chat-Downloader/wiki/Custom-formats) lacks a descriptive list of how to access these variables, assuming that they are available in the first place. 
Knowing what variables are actually available would be helpful for users who want to fine tune their output. I know that at least `message[body]` and `user_name` are variables which are undocumented in the wiki, are there any more?

**Comment by PetterKraabol** [comment]
Twitch Chat Downloader puts JSON data from the Twitch API into a dictionary and uses python's string formatting.

Basically, Python looks up `message[body]` and `user_name` from the API response.
```python
api_response: dict = {
  'user_name': 'TwitchUser',
  'message': {
    'body': 'some message.'
  }
}

# Comment format from settings.json file
command_format: str = '{user_name}: {message[body]}'.

# Format output based on api response
output: str = comment_format.format(**api_response)

print(output) # TwitchUser: some message
```

I can write a detailed list of variables and their description when I have some time.
For now, I'd suggest downloading chat from a short VOD with `--format json` which will give you the raw API data from Twitch.

Example response for video is documented here https://dev.twitch.tv/docs/api/reference/#get-videos

I wrote a [Comment class](https://github.com/PetterKraabol/Twitch-Python/blob/5d05a9428a0737d11b63ad9bd4b2962715c2a648/twitch/v5/models/comment.py) for [Twitch Python](https://github.com/PetterKraabol/Twitch-Python) with typed fields.

```bash
# Comment variables
# Custom variables added by TCD is marked with *
_id: str
created_at: str
updated_at: str
channel_id: str
content_type: str
content_id: str
content_offset_seconds: float
commenter: {
  display_name: str
  _id: str
  name: str
  type: str
  bio: str
  created_at: str
  updated_at: str
  logo: str
  badge: str # *
}
source: str
state: str
message: {
  body: str
  emoticons: [{
    _id: str
    begin: int
    end: int
    emoticon_id: str
    emoticon_set_id: str
  }]
  fragments: [{
    text: str
    emoticon: {
      _id: str
      begin: int
      end: int
      emoticon_id: str
      emoticon_set_id: str
    }
  }]
  is_action: bool
  user_badges: [{
    _id: str
    version: str
  }],
  user_color: str
}
more_replies: bool
timestamp {
  relative: str # *
  absolute: str # *
}
```

**Comment by JAK0723** [comment]
I just tried using the `created_at` video variable, and I got the following error:
```
> tcd -o E:/User/TCD/ --format irc -v 426253207

Pat Stares At Yakuza 6 - The Song of Life
Traceback (most recent call last):
  File "c:\python\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python\Scripts\tcd.exe\__main__.py", line 9, in <module>
  File "c:\python\lib\site-packages\tcd\__init__.py", line 76, in main
    Downloader().videos(Arguments().video_ids)
  File "c:\python\lib\site-packages\tcd\downloader.py", line 162, in videos
    self.video(video)
  File "c:\python\lib\site-packages\tcd\downloader.py", line 127, in video
    with open(output, '+w', encoding='utf-8') as file:
OSError: [Errno 22] Invalid argument: 'E:/User/TCD/AngriestPat/426253207 - 2019-05-18T00:06:02Z Pat Stares At Yakuza 6 - The Song of Life.log'
```
I believe this is caused by the colons in the time.
Would it be possible to sanitize the variables used for output filenames and replace illegal characters with underscores or something?

**Comment by JAK0723** [comment]
This change seems to have fixed the issue with dates, but it appears to have disabled the ability to specify subdirectories.

I would like to have an output format like this `{user_name}/{id} - {created_at} {title}.log`, where the file is placed in the specific `{user_name}` directory. Instead, all downloads are placed in the same folder with the format of `{user_name}{id} - {created_at} {title}.log`, the forward-slash is simply removed.

**Comment by PetterKraabol** [comment]
Thanks! Fixed in 3.0.9
